### ISSUE TITLE
long match arm expression is missing comma with "wrap_match_arms = false"

### ISSUE THREAD

**Issue by oli-obk** [rust-lang-nursery/rustfmt#1127]
in `serde-xml` we have the following match

``` rust
match (&self.state, try!(self.de.ch())) {
            (&Attribute, EmptyElementEnd(_)) => return Ok(None),
            (&Attribute, StartTagClose) => 0,
            (&Attribute, AttributeName(n)) =>
                return Ok(Some(try!(KeyDeserializer::deserialize(try!(self.de.check_utf8(n)))))),
            (&Element, StartTagName(n)) =>
                return Ok(Some(try!(KeyDeserializer::deserialize(try!(self.de.check_utf8(n)))))),
            (&Inner, Text(_)) => 1,
            (&Inner, _) => 4,
            (&Value, EndTagName(_)) => return Ok(None),
            (&Value, Text(txt)) if txt.is_ws() => 3,
            (&Value, Text(_)) => return Ok(Some(try!(KeyDeserializer::value_map()))),
            (&Element, EmptyElementEnd(_)) => 2,
            (&Element, Text(txt)) if txt.is_ws() => 5,
            (&Element, EndTagName(_)) => return Ok(None),
            (&Element, EndOfFile) => return Ok(None),
            _ => unimplemented!(),
        }
```

the two long match arms get their comma removed and there's a trailing empty line with trailing whitespace after each of them.

**Comment by nrc** [comment]
Testing on master, I get:

```
   match (&self.state, try!(self.de.ch())) {
        (&Attribute, EmptyElementEnd(_)) => return Ok(None),
        (&Attribute, StartTagClose) => 0,
        (&Attribute, AttributeName(n)) => {
            return Ok(Some(try!(KeyDeserializer::deserialize(try!(self.de.check_utf8(n))))))
        }
        (&Element, StartTagName(n)) => {
            return Ok(Some(try!(KeyDeserializer::deserialize(try!(self.de.check_utf8(n))))))
        }
        (&Inner, Text(_)) => 1,
        (&Inner, _) => 4,
        (&Value, EndTagName(_)) => return Ok(None),
        (&Value, Text(txt)) if txt.is_ws() => 3,
        (&Value, Text(_)) => return Ok(Some(try!(KeyDeserializer::value_map()))),
        (&Element, EmptyElementEnd(_)) => 2,
        (&Element, Text(txt)) if txt.is_ws() => 5,
        (&Element, EndTagName(_)) => return Ok(None),
        (&Element, EndOfFile) => return Ok(None),
        _ => unimplemented!(),
    }
```

Which looks correct (although I would prefer a semi-colon after the `return`). Are you getting something else?

**Comment by oli-obk** [comment]
My rustfmt.toml:

```toml
max_width = 120
ideal_width = 100
report_todo = "Always"
report_fixme = "Unnumbered"
reorder_imports = true
reorder_imported_names = true
wrap_match_arms = false
match_block_trailing_comma = true
where_trailing_comma = true
```

Now I'm actually getting

```rust
            (&Attribute, AttributeName(n)) =>
                return Ok(Some(try!(KeyDeserializer::deserialize(try!(self.de.check_utf8(n))))))
            (&Element, StartTagName(n)) =>
                return Ok(Some(try!(KeyDeserializer::deserialize(try!(self.de.check_utf8(n))))))
```

(notice the missing comma at the end)

**Comment by topecongiro** [comment]
Closed by #1307.
